Plugin/Controller/Adminhtml/Page/DeletePlugin.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mooore\WordpressIntegrationCms\Plugin\Controller\Adminhtml\Page;
+
+use Magento\Cms\Controller\Adminhtml\Page\Delete as DeleteController;
+use Magento\Cms\Model\PageRepository;
+use Mooore\WordpressIntegrationCms\Model\RemotePageRepository;
+
+class DeletePlugin
+{
+    /**
+     * @var PageRepository
+     */
+    private $pageRepository;
+
+    /**
+     * @var RemotePageRepository
+     */
+    private $remotePageRepository;
+
+    public function __construct(PageRepository $pageRepository, RemotePageRepository $remotePageRepository)
+    {
+        $this->pageRepository = $pageRepository;
+        $this->remotePageRepository = $remotePageRepository;
+    }
+
+    /**
+     * This function calls Wordpress to clear the Magento URL entry on page deletion.
+     *
+     * @param DeleteController $subject
+     * @return void
+     */
+    public function beforeExecute(DeleteController $subject)
```

Comment:
Saving the metadata should only happen when the module has been configured to do so! Also need to check if the API credentials are not empty.

Reply:
The checking of the API credentials happens further down the line in the RemotePageRepository. An error will occur and will be written to the log files for further investigation. Could you explain to me what configuration you would expect to see here?

Reply:
`etc/adminhtml/system.xml`